### PR TITLE
EngiMarket Final PR: The item list, power/atmos/alarm backbone, and the atmos/engi repair vehicles

### PR DESCRIPTION
**Pull request by Robustin**
This is the PR with the core backbone stuff, including the rewards. I've also had to put both repair vehicles in here. They've been reviewed multiple times, weeks ago, and are offer no speed boost/crash effect and are solely there to make it easier to repair an area and/or stabilize the atmosphere and remove fires. 

Keep in mind this PR won't get merged until there's a good deal of test-merging, so  I'd rather start with a long list and cut a few things than limit my options right out of the gate. 

By my estimates that average engineering team should fall between 10-20k per round. If things go to shit immediately then less than 10k, if the team goes above and beyond in a longer round then more than 20k - but on average I want the "luxury" items to feel like a big accomplishment and I'm sure many of you would prefer to keep them rare as well.

But that's why testing is necessary, I guarantee you almost nothing on this list is going down in price, I will revise upwards once I get a feel for how many points engineering will rack up in the chaos of most rounds. 

See the changed file for the list, this PR will be updated to include the map additions once its ready for test_merging. Either that or I'll close the first PR and add the systems to this once I've made sure no more changes are necessary.

### CODE REVIEW

File: code/game/machinery/engi_points_vendor
Comment (kevinz000) [inline code/game/machinery/engi_points_vendor]: Absolutely not.

File: code/game/machinery/engi_points_vendor
Comment (kevinz000) [inline code/game/machinery/engi_points_vendor]: Absolutely, absolutely not.

File: code/game/machinery/engi_points_vendor
Comment (kevinz000) [inline code/game/machinery/engi_points_vendor]: Absolutely not x 1.5

**Comment by Iamgoofball** [comment]
this is not a DM file

**Comment by Robustin** [comment]
The list was old, I cut the decoy armor even after nerfing it into borderline uselessness.

Also Chrono armor is a joke. You drop EVERYTHING, even backpacks, you self-stun upon arrival for so long (even with moderate distances) that an assistant can run up and strip your whole suit before you recover, it also slows you down to make regular movement really awful.

It's pretty much there for funsies, you can achieve 90% of the same effect by monkeying yourself in genetics at roundstart. and crawling into various departments.

File: code/game/machinery/engi_points_vendor
Comment (coiax) [inline code/game/machinery/engi_points_vendor]: The spaces/indentation on this file are upsetting me, please just remove them.

Comment (Jalleo) [inline code/game/machinery/engi_points_vendor]: Holey bees wax I agree please make each one like this
Proc
Name
Path
Number
Also you dont need the x1 after each Item name I believe.

**Comment by Iamgoofball** [comment]
# this isnt a DM file

**Comment by Robustin** [comment]
I'll decide if I'm going to just make this the system PR and close PR#1 if people don't have contributions to make there anyway.

File: code/game/machinery/engi_points_vendor
Comment (Robustin) [inline code/game/machinery/engi_points_vendor]: Formatting got borked when I pulled the list out of the surrounding code. When I decide if this is going to include the map updates/system backbone it'll be fixed

File: code/game/machinery/engi_points_vendor
Comment (KorPhaeron) [inline code/game/machinery/engi_points_vendor]: Pointless since we let people construct them again

File: code/game/machinery/engi_points_vendor
Comment (KorPhaeron) [inline code/game/machinery/engi_points_vendor]: Pointless gas in general

File: code/game/machinery/engi_points_vendor
Comment (KorPhaeron) [inline code/game/machinery/engi_points_vendor]: Balloons indoors are silly and have little to do with their job

File: code/game/machinery/engi_points_vendor
Comment (KorPhaeron) [inline code/game/machinery/engi_points_vendor]: lol

File: code/game/machinery/engi_points_vendor
Comment (KorPhaeron) [inline code/game/machinery/engi_points_vendor]: makes little sense lorewise

File: code/game/machinery/engi_points_vendor
Comment (KorPhaeron) [inline code/game/machinery/engi_points_vendor]: >cars indoors

File: code/game/machinery/engi_points_vendor
Comment (Iamgoofball) [inline code/game/machinery/engi_points_vendor]: i disagree on the indoors fultons, it was good enough for peace walker

also its good for if you need to move a lot of resources fast

File: code/game/machinery/engi_points_vendor
Comment (Iamgoofball) [inline code/game/machinery/engi_points_vendor]: its worth tbh

File: code/game/machinery/engi_points_vendor
Comment (Iamgoofball) [inline code/game/machinery/engi_points_vendor]: >need to repair something
>just teleport in

its good

File: code/game/machinery/engi_points_vendor
Comment (Iamgoofball) [inline code/game/machinery/engi_points_vendor]: yeah bz is useless

Comment (Iamgoofball) [inline code/game/machinery/engi_points_vendor]: doesnt even have sprites

File: code/game/machinery/engi_points_vendor
Comment (KorPhaeron) [inline code/game/machinery/engi_points_vendor]: Why does centcomm have these at all to hand out though

File: code/game/machinery/engi_points_vendor
Comment (Robustin) [inline code/game/machinery/engi_points_vendor]: When was that fixed? Also these are the premium diamond ones and getting diamonds from mining to engineering for boxes is... well its probably never happened.

File: code/game/machinery/engi_points_vendor
Comment (Robustin) [inline code/game/machinery/engi_points_vendor]: did we kill the BZ sprite?

I figured it might work well in the new prototype can for extended release but I don't really care that much one way or another.

Comment (KorPhaeron) [inline code/game/machinery/engi_points_vendor]: It never had one. It also occupies a niche in which it has absolutely zero non antagonist purpose but isn't strong enough to be worth using as an antag. Pure ban bait.

File: code/game/machinery/engi_points_vendor
Comment (Robustin) [inline code/game/machinery/engi_points_vendor]: They have lots of heavy machinery that would make this clutch in something like blob. For naughty antags I'm holding out for the day someone repeats what I did in the video and sends a timed canister full of burning plasma into a secure location. 

Also it could work quite well aboard the new ship. I agree they're kinda silly, I'll wait and see what others say and if they get used during testing.

File: code/game/machinery/engi_points_vendor
Comment (KorPhaeron) [inline code/game/machinery/engi_points_vendor]: In the past couple weeks.

"I can't get X department to work with me so I'll just make my job self contained" has led to everything being boring and safe and overstocked

File: code/game/machinery/engi_points_vendor
Comment (Iamgoofball) [inline code/game/machinery/engi_points_vendor]: Because

Comment (Robustin) [inline code/game/machinery/engi_points_vendor]: Theyre not from Centcom Kor, if you had engaged with the DEEP LORE of IEV's you'd know this was an intergalactic market place that can exchange energy and matter at FTL speeds. The other participants are willing to part with the time-manipulating technology that enabled these sort of transactions in the first place, for a price.

File: code/game/machinery/engi_points_vendor
Comment (Robustin) [inline code/game/machinery/engi_points_vendor]: I've accepted that cars and the hours I put into the may not get merged. I'm going to try and include them for test merging though just to see how people feel when, and if, they show up.

File: code/game/machinery/engi_points_vendor
Comment (Robustin) [inline code/game/machinery/engi_points_vendor]: The deciding factor for me was that engineers can't even access the ORM/RND to get these materials.

For you Kor bby I will cut these and the BZ

**Comment by Bawhoppen** [comment]
It's all just too gimmicky man.

**Comment by SgtxRootbeer** [comment]
you're just now letting us know?

**Comment by Bawhoppen** [comment]
I'm clearly not the only one who thinks its not a suitable feature.

**Comment by Robustin** [comment]
Everything fun about this game is a gimmick in some way Baw. Pretty much every shit role is the one without some gimmick to get people to sign up for it. 

What are the four most popular roles? **Shaft miner, scientist, clown, and chaplain.** 

What do they have in common? Huge fucking gimmicks. Xenobiology is one of the most gimmicky fucking things in the game - you feed monkeys to slimes until you become a changeling or command an army of golems, bears, spiders, and sentient trees? Yet if you tried to remove Xenobio from the game there would be a fucking riot.

It's a 2d Spaceman game thats still growing and gaining followers in 2017, it would've died a long time ago if we didn't have some pretty amazing gimmicks to keep shit fresh. Engineering has been a pretty awful job for a long time now, its a reasonably popular role because you get an amazing roundstart loadout and its one of the best antag positions to get - but the actual JOB has been a joke for years and years. Atmos is a massive farce as well. 

You know the last job to get treatment like this? Shaft miner. Their job, like engineers, was to do a pretty mundane task "for the good of the station" without ever directly benefiting from it. At best they'd get a better drill or something. Then Kor gave them crazy fucking loot for "doing their job", angels wings? Nukeop hardsuits? Becoming a morph? Ghost swords? Then he gave them even crazier fucking loot for stuff totally unrelated to their job. And it was a fucking smash hit. 

I'm trying something similar, except unlike Kor I put an extraordinary burden on myself: No weapons, no combat gear, no crazy supernatural stuff. I had to design, create, and test - for hundreds of hours at this point - a set of rewards that would feel fulfilling and thematic without existing just to help validhunt/grief/murderbone people. In SS13 thats a VERY difficult task, but I think I pulled it off. 

I get why people are uneasy, but I'm confident that when it gets testmerged and people realize this stuff is pretty fun that people will wonder why we didn't have this sooner. Just like any balanced job the goal isn't to give them crazy shit every round, but at the **very least it lets them aspire for it**. So when you roll engineer or atmos tech, don't get antag, and your old "job" is practically done at roundstart, that you actually have a reason to stick around and see how things play out

**Comment by Bawhoppen** [comment]
None of these items seem to really have much substance or deep mechanics behind them. Only like a few items even help with actual repairing, At least the mining rewards vendor starts with basic simple yet useful items. Don't go over the top for the sake of trying to make it interesting, because it just makes it seem gimmicky. Some of your items are cool in their own right, but they don't really help out engineering. Where's the schematic building drone and srugf like that? I think you might need to flesh it all out more

**Comment by Robustin** [comment]
Im talking about the entire lavaland system, the vendor existed before LL and only had "help do your job gear" which was pointless when one haul of minerals is all you needed.

Lavaland became a hit when it got filled with ruins and tendrils packed with crazy loot. I tried to give engineers some stuff, some of it useful for the job (and I pretty much hit on every conceivable idea at some point) and some of it silly, but I stuck by my word that there'd be no weapons or combat gear.

I've given engineering more in this PR than anything else, and I'm open to adding more, but with oranges and others constantly threatening to shut me down I just need to get this stuff testmerged. I wasn't fond of drones because of my experience as an admin around them, they never understand their laws and mostly just try to soft-grief. The ship has a machine builder and maybe that could get fleshed out into something portable at some point but I've had my Ideas thread open for over a month now and got a total of 1-2 actual ideas.

**Comment by Bawhoppen** [comment]
What I meant wasn't player controlled drones, I meant stuff like auto controlled bots that you could make build a pre made schematic for you like factorio robots or something. The engineers need more RCD-like stuff, for quickly building/repairing maybe integrated with the station blueprints whilst leaving a manual old-fashioned way with welders etc available to the crew.

**Comment by Robustin** [comment]
I was screwing around with construction "bombs" that would restore an area of the station to an initial state (not feasible without more map code) and stuff like the mining pod that would load a basic set of machines for a department (sleeper + cryo for medbay, basic RND loadout, basic Tcomms setup, basic brig holding cells), but its simply tabled because I would have to make several new template maps.

The ship construction gun I'm finishing offers an extremely robust tool though. Rebuild APC's, machines, pipes, walls, all remotely and very fast.

**Comment by kevinz000** [comment]
when i play shaft miner my primary source of rewards is from science as no one is going to stop me if the guy who just gave them 5 stacks of each material wants a combat mech man.

**Comment by Supermichael777** [comment]
science sharing the dangerous toys with a peon

File: code/modules/events/ion_storm.dm
Comment (Shadowlight213) [inline code/modules/events/ion_storm.dm]: Dupe. Look 2 lines down.

File: code/modules/events/ion_storm.dm
Comment (Shadowlight213) [inline code/modules/events/ion_storm.dm]: This should probably go into wherever the names for the stock market corporations for cargo are

**Review by Shadowlight213 — CHANGES_REQUESTED** [review]

File: code/modules/events/ion_storm.dm
Comment (Robustin) [inline code/modules/events/ion_storm.dm]: Oranges made me move it here in the first place, /shrug

Comment (Shadowlight213) [inline code/modules/events/ion_storm.dm]: That makes no sense.
@optimumtact this is the file for the ion storm EVENT.
Why should this go here?

Comment (optimumtact) [inline code/modules/events/ion_storm.dm]: no, I told him to use the pick_lists system like the AI ion laws uses and of course he didn't take the time to understand what I actually meant.

Comment (optimumtact) [inline code/modules/events/ion_storm.dm]: that was back when I actually cared about these pr's though.

Comment (Robustin) [inline code/modules/events/ion_storm.dm]: Maybe that was your input and I got you mixed up,  but someone was quite insistent that my proc end up in this file. Where do the cargo stock names come from?

**Comment by Cyberboss** [comment]
f05d157

**Comment by Robustin** [comment]
@Cyberboss ?

**Comment by KorPhaeron** [comment]
@tgstation/commit-access 

If someone doesn't come along and say explicitly "yes I want to merge this and will review the code until it is in a mergable state" I am going to assume nobody wants to and put this PR out of its misery because its been like two weeks of this now

**Comment by SgtxRootbeer** [comment]
@KorPhaeron
I want this merged, does that help?

**Comment by coiax** [comment]
@Robustin fix your action icon conflicts

**Comment by Robustin** [comment]
Kor this PR has been up for 6 days, and I want to have it TEST MERGED before I add the maps, I'll start asking for test merging soon but jesus dude after all the hoops you guys have had me jump through, its not fair to put that kind of deadline on me when I'm still trying to handle everything going on.

**Comment by coiax** [comment]
You can't get a test merge while there are conflicts. Fix em.

**Comment by KorPhaeron** [comment]
I didn't say "merged in 48 hours" or "ready to be merged in 48 hours" I said "someone says they will merge it." You're not on any sort of deadline, the maintainers are.

The concept itself has been talked about for three weeks now.

**Comment by Robustin** [comment]
Can we settle on "someone says the will test merge this"?

Because thats where I was about 10 days ago before I had to atomize it.

**Comment by optimumtact** [comment]
I tried to warn you

**Comment by Robustin** [comment]
Well looks like I get to do my favorite part of coding, politicking. 

Regardless of what happens, the unwillingness to even let testing change people's minds is silly. Hypothetically if this were to be one of the most widely liked PR's ever created, why would anyone still dig in their heels and say "No, fuck that, I won't merge it".

**Comment by optimumtact** [comment]
because it's a still a bad idea

**Comment by Robustin** [comment]
I thought this codebase existed to improve the player experience, not feed coder's egos. I'm a moron for thinking otherwise apparently.

**Comment by Bawhoppen** [comment]
I think if you refined it and made it less over the top rewards and made more basic simple yet powerful tools to actually help engineering out, it would be more likely to get merged. Contrary to what everyone thinks, people will change their minds (forum warriors won't, but actual contributors will).
I know you used the fact we had all kinds of other crazy and OP departments like xenobio, R&D, mining, etc but I don't think that can justify some of these things as is.

**Comment by Robustin** [comment]
@Bawhoppen 

I'm stressing that as I start to speak to maintainers, right now one maintainer has the chance to basically hold hundreds of hours of my work hostage and force me to do whatever they want. I've cut dozens of hours of work already to make this PR more palatable and I'll do plenty more, but now I'm at the mercy of whichever maintainer is willing to at least consider merging ti.

**Comment by dannno** [comment]
>I thought this codebase existed to improve the player experience, not feed coder's egos. I'm a moron for thinking otherwise apparently.

are you having a stroke what are you talking about you r-tard

**Review by Razharas — APPROVED** [review]
I think this deserves to be at least testmerged
Who knows maybe its ok

**Comment by KorPhaeron** [comment]
@Razharas 

Merged or testmerged? The concept isn't going to change substantially during testing, I'd rather have a "yes I like this and will merge eventually" than to continue kicking the can down the road while robustin wastes time

**Comment by Robustin** [comment]
Jesus Kor its almost like the players should have a say in this, and maintainers should be informed by how it plays in reality and not how its theorized in the ascended plane of Coderbus.

**Comment by KorPhaeron** [comment]
We know how it will play. Players will get some toys for doing their job. Players will enjoy that. Whether or not adding another skinner box loot dispenser is good for the overall health of the game is a judgement call maintainers will eventually make.

I don't know where you got the idea that the maintainers are or should be rubber stamps that make the code work for ideas guys to add whatever they want

**Comment by Robustin** [comment]
It's probably the fact I've never had a PR that we knew players would like get turned down before.  I've never been one to add stuff just for popular appeal either. I have always felt it was tragic that engineering and atmos got deeper mechanics than almost anyone else but 0 incentive to explore them. 

So I tried to hit 2 birds with 1 stone. Give engineering an incentive to explore the mechanics that have been ignored for as long as this game has existed, and give engineering a sense of progress that simply doesn't exist right now. 

I would hope that I have some credibility on how to keep mechanics feeling fresh after I've played for so long, since I know exactly what jobs turn into a Wiki Routine and which jobs keep me coming back. To make adjustments I just need to see it in practice first. Will atmos wizards pop out of the woodwork and turn my toughest canister orders into some rote pump work? I can solve those problems but I need to see if they're actually problems first.

**Comment by KorPhaeron** [comment]
Stungloves, movement speed, goofchem, lockboxes, disablers, gun cargo nerfs

Those are the famous times coderbus has "ignored" player opinion for the sake of improving the game off the top of my head

**Comment by Robustin** [comment]
I won't get into a debate about those and whether players really didn't want those changes (too many people just seem to think a straw poll in OOC counts as consensus). 

I'm just saying there are maintainers willing to merge this based upon how they actually see it play out. That may or may not include whether players seem to enjoy it and whether the rewards and the process of earning them doesn't negatively impact "the health of the game". If I weren't this PR's author and were somehow a maintainer (pffft) I would be hesitant to say "I will absolutely merge this at some point in the future" before a single live test had been executed. I can't blame other maintainers for having that position.

My position is that outside of the last PR's conflict, this whole thing is ready for testing minus the shuttle, which is commented out of the rewards right now anyway. It will take a little bit more work for maintainers to verify my position but after several rounds of review there isn't much work left.

**Comment by dannno** [comment]
Every time I look at this PR I distinctly smell sulfur, is that normal

**Comment by Robustin** [comment]
@dannno 

Yea oranges likes to hang out here waiting for an opportunity to strike

**Comment by PKPenguin321** [comment]
>Stungloves, movement speed, goofchem, lockboxes, disablers, gun cargo nerfs

But lockboxes are trash and disablers are still arguably more problematic than tasers

**Comment by KorPhaeron** [comment]
@PKPenguin321 

Not all those decisions were right, but pretending that its without precedent that maintainers would exercise thought independent from the fucking emojis spammed on the PR is nonsense

**Comment by Robustin** [comment]
Yea but Emojis spammed on the PR are not our players, 95% are admins, coders, and a few people who just love watching Github. Anyway even I get exhausted from debating sometimes, point is, this is worth testing and I think it's not a big deal at all. Most rounds these items won't even influence anything that takes place during the round. Most of its aspirational, or it will be adjusted until it is, i.e. you're going to fail or die before getting it 90-95% of the time but at least it gives you a target to shoot for and those 5-10% of the rounds you pull it off are pretty much worth it.

**Comment by Robustin** [comment]
Ok did another round of tests and everything looks good, but #25247 needs merging and #25263 should be done too but Antur needs to sign off on the revisions I made for him.

**Comment by ChangelingRain** [comment]
Sorry dude but literally none of the maintainers want to merge this plus the actual design lead doesn't like it.

**Comment by SgtxRootbeer** [comment]
How the hell did that piece of shit map like cerestation get merged, yet everyone seems to have a vendetta against this?
@iamgoofball wat do, bb?

**Comment by optimumtact** [comment]
because cerestation doesn't take a giant ass shit all over a core game job and can be easily removed.

**Comment by Robustin** [comment]
>Get a maintainer to support merging it after testing
>Joan: LITERALLY 0 MAINTAINERS WANT TO MERGE THIS
>Join IRC and a 2nd maintainer offers to support me pending some changes

HMMMM

Anyway I'm changing the systems off of points toward something more akin to a fabricator but Joan would never miss an opportunity to fuck me over so I'm going to have to make my changes in a closed PR.

**Comment by dannno** [comment]
none of this is about personal vendetta you fucking pussy
can you maybe just understand that some people don't like your idea and think it's shitty?

**Comment by Robustin** [comment]
I've had opponents to every single PR ive ever made, never have I seen such a desperation to even prevent testing.

**Comment by PKPenguin321** [comment]
It's just a pain in the ass to test merge lots of PRs since each one takes time to recompile the game and you have like 8, plus you have to do it for both servers individually

**Comment by SgtxRootbeer** [comment]
@pkpenguin321
So, do we atomize the PR, or not?!?!

**Comment by Robustin** [comment]
PKP its down 4 now, and I've had the other headmins agree to test it as well, but it doesn't matter if everyone loves and it becomes the most loved PR of all time and people build statues in my honor if even 2-3 maintainers have dug their heels in against this. I might still ask for a test if I can catch one of you Headmins when im on but the system will have to be changed if at least for my sanity.

**Pull request closed without merging**
